client/my-sites/post-selector/selector.jsx

```diff
@@ -95,6 +101,11 @@ const PostSelectorPosts = React.createClass( {
 				requestedPages: [ 1 ]
 			} );
 		}
+
+		if ( this.props.posts !== nextProps.posts ) {
```

Comment:
>is performing this comparison a bit intensive due to posts being an array of large objects?

If it were a deep equality check (`_.isEqual`), then yes it would be, but this is simply comparing references, made possible only by the change in f94a36a to memoize and return the same result from the selector until state changes.